src/pages/index.js

```diff
@@ -51,55 +73,154 @@ const popupCardForm = new PopupWithForm(editCardModal, handleCardFormSubmit);
 popupProfileForm.setEventListeners();
 popupCardForm.setEventListeners();
 
-//Section Class
-const section = new Section(
-  { items: initialCards, renderer: renderCard },
-  cardListEl
+//PopupWithProfilePicForm
+const popupProfileImageForm = new PopupWithProfilePicForm(
+  profileImageModal,
+  handleProfileImageFormSubmit
 );
-section.renderItems();
+popupProfileImageForm.setEventListeners();
+
+//PopupWithDelete Class
+const popupWithDeleteCard = new PopupWithDelete(
+  deleteCardModal,
+  handleDeleteCardFormSubmit
+);
+popupWithDeleteCard.setEventListeners();
 
 //UserInfo Class
-const userClass = new UserInfo({
+const userInfo = new UserInfo({
   nameContainer: profileName,
   descriptionContainer: profileDescription,
 });
 
+//Api Class
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "1755563a-df02-4ecd-aa91-5c5958bedadf",
+    "Content-Type": "application/json",
+  },
+});
+
+//Section Class
+const section = new Section({ renderer: renderCard }, cardListEl);
+
+api.getInitialCards().then((cardItems) => {
+  section.renderItems(cardItems);
+});
+
+api.getProfileInfo().then((res) => {
+  profilePic.src = res.avatar;
+
+  return userInfo.setUserInfo({
+    nameInput: res.name,
+    descriptionInput: res.about,
+  });
+});
+
+// initialCards.forEach((card) => {
+//   api.addCard(card);
+// });
+
 //
 //Functions
 function fillProfileInputs() {
-  const inputs = userClass.getUserInfo();
+  const inputs = userInfo.getUserInfo();
 
   profileNameInput.value = inputs.profileName;
   profileDescriptionInput.value = inputs.profileDescription;
 }
 
 function handleProfileFormSubmit(data) {
-  userClass.setUserInfo({
-    nameInput: data.title,
-    descriptionInput: data.description,
-  });
-
-  popupProfileForm.close();
+  api
+    .updateProfileInfo({
+      name: data.title,
+      about: data.description,
+    })
+    .then(() => {
+      userInfo.setUserInfo({
+        nameInput: data.title,
+        descriptionInput: data.description,
+      });
+      popupProfileForm.close();
+      setTimeout(() => {
```

Comment:
Needs correction: you don't need the timeouts here. Change button state in `finally` block, otherwise the button text won’t change back if an error occurred

Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Fix that everywhere